spec/site_update_priority_spec.rb

```diff
@@ -0,0 +1,52 @@
+# frozen_string_literal: true
+
+require 'spec_helper'
+
+describe SFRest::Site_update_priority do
+  before :each do
+    @conn = SFRest.new "http://#{@mock_endpoint}", @mock_user, @mock_pass
+  end
+
+  describe '#get_current_site_update_priority' do
+    path = '/api/v1/site-update-priority'
+
+    it 'calls the site update priority get endpoint' do
+      stub_request(:any, /.*#{@mock_endpoint}.*#{path}/)
+        .with(headers: @mock_headers)
+        .to_return { |request| { body: { uri: request.uri, body: request.body, method: request.method }.to_json } }
+      res = @conn.site_update_priority.current_update_priority
+      uri = URI res['uri']
+      expect(uri.path).to eq path
+      expect(res['method']).to eq 'get'
+    end
+  end
+
+  describe '#set_site_update_priority' do
+    path = '/api/v1/site-update-priority'
+
+    it 'calls the site update priority put endpoint' do
+      stub_request(:any, /.*#{@mock_endpoint}.*#{path}/)
+        .with(headers: @mock_headers)
+        .to_return { |request| { body: { uri: request.uri, body: request.body, method: request.method }.to_json } }
+      payload = [22, 28, 24]
+      res = @conn.site_update_priority.change_update_priority(payload)
+      uri = URI res['uri']
+      expect(uri.path).to eq path
+      expect(res['method']).to eq 'put'
+    end
+  end
+
+  describe '#reset_site_update_priority' do
+    path = '/api/v1/site-update-priority'
+
+    it 'calls the site update priority delete endpoint' do
+      stub_request(:any, /.*#{@mock_endpoint}.*#{path}/)
+        .with(headers: @mock_headers)
+        .to_return { |request| { body: { uri: request.uri, body: request.body, method: request.method }.to_json } }
+      res = @conn.site_update_priority.reset_update_priority
+      uri = URI res['uri']
+      expect(uri.path).to eq path
+      expect(res['method']).to eq 'delete'
```

Comment:
Similar question to above: is it worth asserting that the body that gets passed into the service is empty for DELETE?

I'm not sure I'm reading this right, but it looks like basically: we set up a mock service instead of an actual factory API.  That mock service accepts requests on the same path as the actual factory API, and it always returns a payload that tells you what was passed in: the URI, method and body.   We might be able to  make more use of the body in assertions generally (if that works).

Reply:
To be honest, I think it's fine since on the API side we don't care if anything's passed there.